src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,110 @@
+package duke.task;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a list of Tasks for the Duke program.
+ */
+public class TaskList implements Serializable {
+    protected ArrayList<Task> tasks;
+
+
+    /**
+     * Instantiates a TaskList with a specific List.
+     * @param tasks the list to instantiate the TaskList with.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = new ArrayList<>(tasks);
+    }
+
+    /**
+     * Instantiates a TaskList with no existing tasks.
+     */
+    public TaskList() {
+        this(new ArrayList<>());
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     * @param task the task to be added
+     */
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Marks a certain task in the TaskList as done.
+     * @param index the 0-based index of the task to be marked as done
+     * @return the task that was marked as done
+     */
+    public Task markTaskAsDone(int index) {
+        Task task = tasks.get(index);
+        task.setDone(true);
+        return task;
+    }
+
+    /**
+     * Returns the task at a certain 0-based index of the TaskList.
+     * @param index the 0-based index of the task to be returned
+     * @return the task at the 0-based index
+     */
+    public Task get(int index) {
+        return tasks.get(index);
+    }
+
+    /**
+     * Removes a task at the 0-based index of the TaskList
+     * @param index the 0-based index of the task to be removed
+     * @return the task that was removed
+     */
+    public Task remove(int index) {
+        return tasks.remove(index);
+    }
+
+    /**
+     * Returns a TaskList where all the terms' descriptions contain the search term
+     * @param term the String to search for
+     * @return the filtered TaskList
+     */
+    public TaskList find(String term) {
+        ArrayList<Task> newList = new ArrayList<>(tasks);
+        newList.removeIf((task -> !task.getDescription().contains(term)));
+        return new TaskList(newList);
+    }
+
+    /**
+     * Returns the size of the TaskList.
+     * @return the size of the TaskList
+     */
+    public int size() {
```

Comment:
Very minor nitpick but this could be named getSize() instead

Reply:
I see where you're coming from, but I thought it would be better to imitate the naming of the method in ArrayList.